packages/testing/src/playwright/unstable/page-objects/checkout.ts

```diff
@@ -0,0 +1,47 @@
+import type { EnhancedPage } from './app';
+import { common } from './common';
+
+export const createCheckoutPageObject = (testArgs: { page: EnhancedPage }) => {
+  const { page } = testArgs;
+  const self = {
+    ...common(testArgs),
+    waitForMounted: (selector = '.cl-checkout-root') => {
+      return page.waitForSelector(selector, { state: 'attached' });
+    },
+    fillTestCard: async () => {
+      await self.fillCard({
+        number: '4242424242424242',
+        expiration: '1234',
+        cvc: '123',
+        country: 'United States',
+        zip: '12345',
+      });
+    },
+    fillCard: async (card: { number: string; expiration: string; cvc: string; country: string; zip: string }) => {
+      const frame = page.frameLocator('iframe[src*="elements-inner-payment"]');
+      await frame.getByLabel('Card number').fill(card.number);
+      await frame.getByLabel('Expiration date').fill(card.expiration);
+      await frame.getByLabel('Security code').fill(card.cvc);
+      await frame.getByLabel('Country').selectOption(card.country);
+      await frame.getByLabel('ZIP code').fill(card.zip);
+    },
+    continueWithSavedCard: async () => {
+      await page
+        .locator('.cl-disclosureRoot')
```

Comment:
do we need to select the disclosureRoot here vs just selecting the button trigger?

Reply:
The button to pay with a saved card and the button to pay with a new card use the same labels, so we need to distinguish between them. This selector is "select the subscribe/pay button that's a child of a cl-disclosureRoot containing a button with the name Payment Methods" which distinguishes it from the `Pay $x.xx` button within the Add new payment method form (which is still in the DOM, but not visible). I wanted this method to work even if the `Add new payment method` disclosure was open.

![Screen Shot 2025-05-01 at 10 49 26 AM](https://github.com/user-attachments/assets/b3f10629-758b-4c26-ba3a-695fe424aa27)